File: api/src/main/java/com/rxdroid/api/github/provider/GitHubRepositoryProvider.java
package com.rxdroid.api.github.provider;

import android.support.annotation.NonNull;
import com.rxdroid.api.github.GitHubApi;
import com.rxdroid.api.github.model.GitHubRepoModel;
import dagger.Reusable;
import io.reactivex.Observable;
import java.util.List;
import javax.inject.Inject;
import retrofit2.Response;

@Reusable
public class GitHubRepositoryProvider extends GitHubApiProvider implements ApiProvider<List<GitHubRepoModel>> {

    @Inject
    public GitHubRepositoryProvider(final GitHubApi gitHubApi) {
        super(gitHubApi);
    }

    @Override
    public Observable<Response<List<GitHubRepoModel>>> loadBySearchValue(@NonNull final String searchValue) {
        return getGitHubApi().getReposForUser(searchValue);
    }
}

File: api/src/main/java/com/rxdroid/api/github/provider/GitHubRepositoryProvider.kt
package com.rxdroid.api.github.provider

import com.rxdroid.api.github.GitHubApi
import com.rxdroid.api.github.model.GitHubRepoData
import dagger.Reusable
import io.reactivex.Observable
import retrofit2.Response
import javax.inject.Inject

@Reusable
class GitHubRepositoryProvider @Inject
constructor(gitHubApi: GitHubApi) : GitHubApiProvider(gitHubApi), ApiProvider<List<GitHubRepoData>> {

    override fun loadBySearchValue(searchValue: String): Observable<Response<List<GitHubRepoData>>> {
        return gitHubApi.getReposForUser(searchValue)
    }
}
